### PR TITLE
fix: bump nodejs-resolver to fix path issue with chinese characaters

### PR DESCRIPTION
**Pull request by bvanjoi**
This PR upgrade the version of `nodejs_resolver`, it brings following change:

1. Optimized handling of `extendsions`, which reduces content usage and optimizes execution speed.  relative: https://github.com/web-infra-dev/nodejs_resolver/issues/157
2. Fixed the paths map match in tsconfig.
    - some test: https://github.com/web-infra-dev/nodejs_resolver/pull/172/files#diff-9bfc51173c154df5ab54a96fc510a2c449c578905e2b9be080b19450792d5236R193
4. fix some string slice in utf8. 
    - test case: https://github.com/web-infra-dev/nodejs_resolver/commit/dc99710c2612f01f1ecbf99edfb1b164c6e8e19e#diff-24975acf43b4bc39c4c8b57246e4bfbdc0c76428c86ed68f1db0e1b10a6db277R1093-R1101
    - close https://github.com/web-infra-dev/rspack/issues/2546
5. support `resolve.fullySpecified`, close https://github.com/web-infra-dev/rspack/issues/1125

### CODE REVIEW

**Comment by bvanjoi** [comment]
!canary

**Comment by bvanjoi** [comment]
!canary

**Comment by h-a-n-a** [comment]
!canary

**Comment by h-a-n-a** [comment]
!canary

**Comment by bvanjoi** [comment]
!canary

**Review by h-a-n-a — APPROVED** [review]

**Review by ahabhgk — DISMISSED** [review]

File: packages/rspack/src/config/types.ts [hunk 1]
```diff
@@ -484,13 +485,13 @@ export type OptimizationRuntimeChunk =
 	| ("single" | "multiple")
 	| boolean
 	| {
-			name?: string | Function;
-	  };
+		name?: string | Function;
```
Comment (hardfist) [inline packages/rspack/src/config/types.ts:488]: why we have lots format change here, do we check format in ci? @IWANABETHATGUY

**Review by ahabhgk — APPROVED** [review]

**Review by hyf0 — APPROVED** [review]

**Pull request merged**